packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts

```diff
@@ -313,16 +314,21 @@ describeNoCompat("GC inactive nodes tests", (getTestObjectProvider) => {
 				// Wait for inactive timeout. This will ensure that the unreferenced data store is inactive.
 				await waitForInactiveTimeout();
 
-				// Load a non-summarizer container from the above summary that uses the mock logger.
+				// Load a non-summarizer container from the above summary that uses the mock logger. This container has to
+				// be in "write" mode for GC to initialize unreferenced nodes from summary.
 				const container2 = await provider.loadTestContainer(
 					{
-						// AB#3982 track work to removing this exception using simulateReadConnectionUsingDelay
-						simulateReadConnectionUsingDelay: false,
 						...testContainerConfig,
 						loaderProps: { logger: mockLogger },
 					},
 					{ [LoaderHeader.version]: summaryVersion1 },
 				);
+				const defaultDataStoreContainer2 = await requestFluidObject<ITestDataObject>(
+					container2,
+					"/",
+				);
+				defaultDataStoreContainer2._root.set("mode", "write");
```

Comment:
What's the reason behind inactive telemetry not being able to fire for read mode clients?